dash.el

```diff
@@ -1170,20 +1170,28 @@ returns the header value, but only after seeing at least one
 other value (the body)."
   (--partition-by-header (funcall fn it) list))
 
-(defun -partition-after-pred (pred list)
+(defmacro -partition-after-pred (pred list)
```

Comment:
It's not possible to turn a function into a macro with the same name, as that will break its existing callers. Also, Dash follows the naming convention where functions start with a single dash (e.g. `-foo`) and their macro counterparts start with two dashes (e.g. `--foo`).

I therefore suggest that, if you deem a macro is necessary, you provide that separately, and optionally reimplement the function in terms of the macro.

Reply:
The macro version should then be anaphoric in the predicate to follow the rest.

Personally these days I prefer to write the macros in terms of a function, the performance impact seems ignorable (until someone finds it slow enough).

Reply:
> (until someone finds it slow enough)

;)

Reply:
Hi! Thank you very much for the review! I'll rework it accordingly and be back,

Reply:
Now it is a function again, and a new anaphoric macro has been added:
  `(defun -partition-after-pred (pred list) ...)`
  `(defmacro --partition-after-pred (form list) ...)`